backend/fms_core/resources/sample.py

```diff
@@ -177,6 +181,13 @@ def import_obj(self, obj, data, dry_run):
             individual = None
             errors["individual"] = ValidationError(e.messages.pop(), code="invalid")
 
+        if data["Project"]:
+            project_name = get_normalized_str(data, "Project")
+            if not Project.objects.filter(name=project_name).exists():
```

Comment:
Another comment about readability :P Do we need to do a 'not' ? why dont you do 
"if Project.objects.filter(name=project_name).exists():
     self.project_object ... 
 else:
    errors["project'] = ValidationError .... "

Reply:
EDIT: I re-read the lines, and I rather suggest:

try:
   self.project = Project.objects.get(..... )
except ObjectDoesNotExist:
   errors["project'] = ...  
   
More straightforward to me (+ possibly save 1 db hit :P )

Reply:
Good catch!!